app/charts/shared/interaction/tooltip-box.tsx

```diff
@@ -348,3 +366,38 @@ const mkTriangle = (p: TooltipPlacement) => {
       };
   }
 };
+
+const toolTipXBoundary = (
+  value: number,
+  tooltipWidth: number,
+  chartWidth: number
+) => {
+  return Math.max(
+    tooltipWidth / 2 - TRIANGLE_SIZE,
+    Math.min(chartWidth - tooltipWidth / 2 + TRIANGLE_SIZE, value)
+  );
+};
+
+const triangleXPos = (
+  value: number,
+  tooltipWidth: number,
+  chartWidth: number
+): number => {
+  const condition = chartWidth - tooltipWidth / 2 + TRIANGLE_SIZE < value;
```

Comment:
Nit: A `condition` for what? 😄 I think we lack an intent in the variable name here. Is this about `isOverflowing`?